Table docs: updates to further clarify alignment (#1246)
hashicorp/design-system: website/docs/components/table/partials/code/how-to-use.md

## How to use this component

!!! Info

This component takes advantage of the `sort-by` helper provided by [ember-composable-helpers](https://github.com/DockYard/ember-composable-helpers).

> !!!

Author: I want this information up-front for a few reasons; one is that many apps are already this addon in their applications. So the idea is, this thing is probably already familiar to you. But there was other feedback that the previous iteration felt too wordy, so I tried to strip it down to the bare minimum. Does that help clarify? WDYT?

> !!!

Reviewer: I like the stripped-down version. ✨ I'm more curious about its placement in the overall hierarchy of the docs because it's only needed for sortable tables, correct? If so, it feels like there's a more natural fit for this content within the section on sortable tables.

> !!!

Reviewer 2: @MelSumner not a blocker, but I strongly agree with @heatherlarsen: this is too prominent, and I think should be moved under the sortable tables examples where is more in context

**Important**

For clarity, there are a couple of important points to note here:

- provide a `@columns` argument (see [Component API](#component-api) for details about its shape)
- use the `.data` key to access the `@model` record content (it’s yielded as `data`)

> !!!

Author: Let's save this one for a separate PR, since it's not a change introduced by this PR and is out of scope for the associated JIRA ticket(s).

> You can optionally also indicate that the column defined in `@sortBy` should be pre-sorted in a descending order, rather than the default ascending order by passing in `@sortOrder='desc'`.

Reviewer: ```suggestion
By default, the sort order is set to ascending. To indicate that the column defined in `@sortBy` should be pre-sorted in descending order, pass in `@sortOrder='desc'`.
```

> To implement a custom sort callback on a columns, (1) add a custom function as the value for `sortingFunction` in the column hash, and (2) include a custom `onSort` action in your table invocation to track the sorting order and use it in the custom sorting function. This is useful for cases where the key might not be A-Z or 0-9 sortable by default, i.e., status, and you’re otherwise unable to influence the shape of the data in the model. Here is an example, code truncated for clarity:

Reviewer: ```suggestion
To implement a custom sort callback on a column:

1. add a custom function as the value for `sortingFunction` in the column hash,
2. include a custom `onSort` action in your table invocation to track the sorting order and use it in the custom sorting function. 

This is useful for cases where the key might not be A-Z or 0-9 sortable by default, e.g., status, and you’re otherwise unable to influence the shape of the data in the model. 

_The code has been truncated for clarity._
```

// and updates the tracked sort order values accordingly
@action
customOnSort(_sortBy, sortOrder) {
  this.customSortOrderForStatus = sortOrder;
}
```

### Density

To create a condensed or spacious table, add `@density` to the table's invocation. Note that it only affects the table body, not the table header.

```handlebars{data-execute=false}
<Hds::Table
  @model={{this.model.data}}
  @columns={{array
    (hash key="artist" label="Artist" isSortable=true)
    (hash key="album" label="Album" isSortable=true)
    (hash key="year" label="Release Year")
  }}
  @density="short"
>
  <:body as |B|>
    <B.Tr>
      <B.Td>{{B.data.artist}}</B.Td>
      <B.Td>{{B.data.album}}</B.Td>
      <B.Td>{{B.data.year}}</B.Td>
    </B.Tr>
  </:body>
</Hds::Table>
```

### Alignment

#### Vertical alignment

To indicate that the table's content should have a middle vertical-align, use `@valign` in the table's invocation.

!!! Info

Note that vertical-align only applies to inline, inline-block and table-cell elements: you can't use it to vertically align block-level elements ([see MDN reference](https://developer.mozilla.org/en-US/docs/Web/CSS/vertical-align)).

> If you have more than just text content in the table cell, you'll want to wrap that content in a flex box and style accordingly.

Reviewer: This content feels more related to the heading and example below than it does to the informational message above. What about moving it under the heading "Vertical alignment with additional cell content"?

Author: The goal was to explain why vertical align alone won't work, and what the consumer should do instead. WDYT?

Reviewer: Right, I understand that. What I'm suggesting is that the content within the info banner feels more related to the code example under the heading "Vertical alignment with additional cell content" than where it's placed now.

It feels like there's opportunity to move it here for a more natural flow of content:
<img width="879" alt="Screenshot 2023-03-14 at 1 15 16 PM" src="https://user-images.githubusercontent.com/8553306/225126104-b7b58df2-8a6e-4d22-9441-b5c0f75485f3.png">

Author: I see what you mean. I've updated it per your suggestion.

    (hash key="year" label="Release Year")
  }}
  @valign="middle"
>
  <:body as |B|>
    <B.Tr>
      <B.Td>{{B.data.artist}}</B.Td>
      <B.Td>{{B.data.album}}</B.Td>
      <B.Td>{{B.data.year}}</B.Td>
    </B.Tr>
  </:body>
</Hds::Table>
```

#### Vertical Alignment with additional cell content

```handlebars{data-execute=false}
<Hds::Table
  @model={{this.model.data}}
  @columns={{array
    (hash key="artist" label="Artist" isSortable=true)
    (hash key="album" label="Album" isSortable=true)
    (hash key="year" label="Release Year")
  }}
  @valign="middle"
>
  <:body as |B|>
    <B.Tr>
      <B.Td>
        <div class="my-flexbox">
          <FlightIcon @name="headphones" /> {{B.data.artist}}
        </div>
      </B.Td>
      <B.Td>{{B.data.album}}</B.Td>
      <B.Td>{{B.data.year}}</B.Td>
    </B.Tr>
  </:body>
</Hds::Table>
```

#### Text Alignment

To create a column that has right-aligned content, set `@align` to `right` on both the column's header and cell.

```handlebars{data-execute=false}
<Hds::Table
  @model={{this.model.data}}
  @columns={{array
    (hash key="artist" label="Artist" isSortable=true)
    (hash key="album" label="Album" isSortable=true)

>     (hash key="year" label="Release Year" align=right)

Author: ```suggestion
    (hash key="year" label="Release Year" align="right")
```
